jormungandr/src/client.rs

```diff
@@ -57,45 +56,46 @@ pub fn handle_input(
             );
         }
         ClientMsg::GetPeers(handle) => {
-            let fut = handle.async_reply(get_peers(&task_data.topology));
+            let fut = async move {
+                let peers = get_peers(&task_data.topology).await;
+                handle.reply(peers);
+            };
             let logger = info.logger().new(o!("request" => "GetPeers"));
 
-            info.spawn(
+            info.spawn_failable_std(
                 "get peers",
-                Timeout::new(fut, Duration::from_secs(PROCESS_TIMEOUT_GET_PEERS)).map_err(
-                    move |e| {
-                        error!(
-                            logger,
-                            "request timed out of failed unexpectdly";
-                            "error" => ?e,
-                        );
-                    },
-                ),
+                timeout(Duration::from_secs(PROCESS_TIMEOUT_GET_PEERS), fut).map_err(move |e| {
+                    error!(
+                        logger,
+                        "request timed out of failed unexpectdly";
+                        "error" => ?e,
+                    );
+                }),
             );
         }
         ClientMsg::GetHeaders(ids, handle) => {
             info.timeout_spawn_failable_std(
                 "GetHeaders",
                 Duration::from_secs(PROCESS_TIMEOUT_GET_HEADERS),
-                handle_get_headers(task_data.clone(), ids, handle),
+                handle_get_headers(task_data.clone(), ids, handle.into_03()),
```

Comment:
This could be a good time to change the handle type to the 0.3 API in the messages, as well.

Reply:
Did it and also removed 0.2 API `ReplyStreamHandle`.